src/client/java/dev/creesch/WebchatClient.java

```diff
@@ -92,5 +102,28 @@ public void onInitializeClient() {
 
             webInterface.shutdown();
         });
+
+        // For the client in fabric there are no events to listen for other players joining or leaving the server.
+        // In order to do that a mixin would be needed to directly access minecraft.
+        // Considering that reading the player list is a fairly low impact operation it simply
+        // works out to just send clients updates every couple of ticks.
+        ClientTickEvents.END_CLIENT_TICK.register(client -> {
+            // Only used to send player list updates. So a client is needed and a world (on a server)
+            if (client == null || client.world == null) {
+                return;
+            }
+
+            tickCounter++;
+            // Send update every 80 ticks (~4 seconds depending on circumstances in game)
+            // Not fast but should be good enough for most use cases.
+            if (tickCounter % 80 == 0) {
```

Comment:
Nit: Could you do this by checking if the player list is different from before and only send if it is? Maybe then you can increase the frequency.

Reply:
I did consider this, but then we are doing comparisons on the minecraft client side on a much more frequent basis. I suppose I could keep a very simple playerId list and see if that is still the same. I'll think about it for a bit.